File: Concepts/Assignment2/Assignment4/student.py
class Student:
    def __init__(self,  name, student_number, birthday, major, gpa):
        self.name = name
        self.student_number = student_number
        self.birthday = birthday
        self.major = major
        self.gpa = gpa

    def on_probation(self):
        pass

    def __str__(self):
        probation = "Yes" if self.on_probation() else "No"
        return f"{self.name} - {self.student_number} - Probation: {probation}"

File: Concepts/Assignment1/StudentBST/src/Student.java
import java.time.LocalDate;

public abstract class Student implements Comparable<Student>{
    private final String name;
    private final String studentNumber;
    private final LocalDate birthday;
    private final String major;
    private final double gpa;

    public Student(String name, String studentNumber, LocalDate birthday, String major, double gpa) {
        this.name = name;
        this.studentNumber = studentNumber;
        this.birthday = birthday;
        this.major = major;
        this.gpa = gpa;
    }
    public String getName() {
        return name;
    }
    public String getStudentNumber() {
        return studentNumber;
    }
    public LocalDate getBirthday() {
        return birthday;
    }
    public String getMajor() {
        return major;
    }
    public double getGpa() {
        return gpa;
    }
    public abstract boolean isOnProbation();

    @Override
    public int compareTo(Student other) {
        return this.getStudentNumber().compareTo(other.getStudentNumber());
    }
    @Override
    public String toString() {
        return getName() + " " + getStudentNumber() + " " +
                (isOnProbation() ? "PROBATION" : "OK");
    }


}
